ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/BlockTransactionSelector.java

```diff
@@ -342,14 +331,44 @@ private TransactionSelectionResult evaluateTransaction(
     return TransactionSelectionResult.CONTINUE;
   }
 
+  private boolean transactionCurrentPriceBelowMin(final Transaction transaction) {
+    // Here we only care about EIP1159 since for Frontier and local transactions the checks
+    // that we do when accepting them in the pool are enough
+    if (transaction.getType().supports1559FeeMarket()
+        && !pendingTransactions.isLocalSender(transaction.getSender())) {
+
+      // For EIP1559 transactions, the price is dynamic and depends on network conditions, so we can
+      // only calculate at this time the current minimum price the transaction is willing to pay
+      // and if it is above the minimum accepted by the node.
+      // If below we do not delete the transaction, since when we added the transaction to the pool,
+      // we assured sure that the maxFeePerGas is >= of the minimum price accepted by the node
+      // and so the price of the transaction could satisfy this rule in the future
+      final Wei currentMinTransactionGasPriceInBlock =
+          feeMarket
+              .getTransactionPriceCalculator()
+              .price(transaction, processableBlockHeader.getBaseFee());
+      if (minTransactionGasPrice.compareTo(currentMinTransactionGasPriceInBlock) > 0) {
+        LOG.debug(
```

Comment:
Volume could be very high, perhaps LOG.trace?

Reply:
done